.github/workflows/build-binaries.yaml

```diff
@@ -0,0 +1,163 @@
+name: Build, Test and Publish Pyinstaller Binaries
+
+on:
+  workflow_call:
+    inputs:
+      production_release:
+        required: true
+        type: string
+      python_version:
+        required: true
+        type: string
+      release_version:
+        required: false
+        type: string
+
+jobs:
+  build-binaries:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false # Continue with other matrix jobs even if one fails
+      matrix:
+        # macos-14 is the Apple Silicon M1 runner (mac os 14)
+        # macos-13 is the last available Intel Mac runner (mac os 13)
+        # See https://github.com/actions/runner-images?tab=readme-ov-file#available-images
+        os: [ubuntu-22.04, ubuntu-22.04-arm, windows-latest, macos-13, macos-14]
+
+    steps:
+      - name: Set signing condition
+        id: signing
+        run: |
+          # Allow signing on:
+          # 1. Main branch non-PR events when production_release is true OR
```

Comment:
Is there any reason to disable signing when on `main` branch and `production_release` is false?

Reply:
This conditional migrated from algokit-cli, https://github.com/algorandfoundation/algokit-cli/pull/604 - in case of puya i suppose it makes sense to allow running codesign on main when prod release is false as well. Modified to allow running codesigning on any push to main